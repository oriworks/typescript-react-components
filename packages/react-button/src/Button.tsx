import * as React from 'react'
import styled from 'styled-components'

export interface LabelProps {
  labelWidth?: number
}

export interface InputWithLabelProps extends LabelProps {
  id?: string
  label?: string
}

export interface InputWithoutLabelProps extends LabelProps {
  id: string
  label: string
}

export type InputLabelProps = InputWithLabelProps | InputWithoutLabelProps

export interface InputProps {
  name?: string
  type?: string
}

const Wrapper = styled.div`
  display: flex;
  margin: 10px;
`

const Label = styled.label`
  color: red;
  margin-right: 10px;
  font-weight: bold;
  width: ${({ labelWidth = 100 }: LabelProps) => labelWidth}px;
`

const NativeInput = styled.input`
  width: 100%;
`

export const Button: React.FC<InputProps & InputLabelProps> = ({ label, id, labelWidth, ...rest }: InputProps & InputLabelProps) => (
  <Wrapper>
    { label && (
      <Label labelWidth={labelWidth} htmlFor={id}>
        {label}:
      </Label>
    )}
    <NativeInput id={id} {...rest} />
  </Wrapper>
)

Button.defaultProps = {
  type: 'text'
}
