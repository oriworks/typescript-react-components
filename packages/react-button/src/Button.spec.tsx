import * as React from 'react'
import { render } from '@testing-library/react'

import { Button } from './Button'

describe('Button', () => {
  test('should match snapshot and styles for default props', () => {
    // Act
    const { container: component } = render(<Button />)

    // Assert
    expect(component).toMatchSnapshot()
  })

  test('should match snapshot with label', () => {
    // Act
    const { container: component } = render(<Button id="test" label="Name" />)

    // Assert
    expect(component).toMatchSnapshot()
  })
})
