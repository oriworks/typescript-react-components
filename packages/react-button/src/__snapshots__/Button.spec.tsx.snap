// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot and styles for default props 1`] = `
<div>
  <div
    class="sc-AxjAm fzDdBB"
  >
    <input
      class="sc-AxiKw hSLCUF"
      type="text"
    />
  </div>
</div>
`;

exports[`Button should match snapshot with label 1`] = `
<div>
  <div
    class="sc-AxjAm fzDdBB"
  >
    <label
      class="sc-AxirZ dLEKRn"
      for="test"
    >
      Name
      :
    </label>
    <input
      class="sc-AxiKw hSLCUF"
      id="test"
      type="text"
    />
  </div>
</div>
`;
